subsys/fw_info/fw_info.c

```diff
@@ -66,14 +66,17 @@ static bool ext_api_satisfies_req(const struct fw_info_ext_api * const ext_api,
 		&& ((ext_api->ext_api_flags & req_flags) == req_flags));
 }
 
+#ifndef CONFIG_EXT_API_PROVIDE_EXT_API_REQUIRED
```

Comment:
Could you explain why this check is here? The way the EXT_API's work are kinda magic, so IMO we should document as much as we can in the code to make it easier to understand.